File: 0633-sum-of-square-numbers/0633-sum-of-square-numbers.c
bool is_sqr(int n) {
    double sqrt_n = sqrt(n);
    return sqrt_n == (int)sqrt_n;
}

bool judgeSquareSum(int c) {
    if (c <= 2) return 1;
    while (!(c % 4)) c /= 4;
    if (c % 8 == 7) return 0;
    int sqrt_c = sqrt(c);
    if (is_sqr(c)) return 1;
    for (int i = 1; i <= sqrt_c; i++) if (is_sqr(c - i * i)) return 1;
    return 0;
}

File: 0633-sum-of-square-numbers/0633-sum-of-square-numbers.java
class Solution {
    public boolean judgeSquareSum(int c) {
        long l = 0, r = (long)Math.sqrt((double)c);
        while (l <= r) {
            if (l * l + r * r == c) return true;
            else if (l * l + r * r > c) r--;
            else l++;
        }
        return false;
    }
}